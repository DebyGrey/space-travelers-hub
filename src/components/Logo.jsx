import logo from '../assets/logo.png';

const Logo = () => (
  <div className="flex justify-start items-center gap-x-4">
    <img src={logo} className="h-[60px] " alt="logo" />
    <h1 className="text-3xl font-bold">
      Space Travelers&apos; Hub
    </h1>
  </div>
);
export default Logo;
