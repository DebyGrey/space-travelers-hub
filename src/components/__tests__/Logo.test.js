import { render, screen } from "@testing-library/react";
import Logo from "../Logo";
import logo from "../../assets/logo.png";

describe("Logo", () => {
  test("renders the logo image", () => {
    render(<Logo />);
    const logoImage = screen.getByAltText("logo");
    expect(logoImage).toBeInTheDocument();
    expect(logoImage).toHaveAttribute("src", logo);
    expect(logoImage).toHaveClass("h-[60px]");
  });

  test("renders the title", () => {
    render(<Logo />);
    const titleElement = screen.getByText("Space Travelers' Hub");
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveClass("text-3xl");
    expect(titleElement).toHaveClass("font-bold");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
