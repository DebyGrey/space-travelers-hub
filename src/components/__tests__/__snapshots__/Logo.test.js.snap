// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-start items-center gap-x-4"
  >
    <img
      alt="logo"
      class="h-[60px] "
      src="logo.png"
    />
    <h1
      class="text-3xl font-bold"
    >
      Space Travelers' Hub
    </h1>
  </div>
</DocumentFragment>
`;
